business/payment.go

```diff
@@ -0,0 +1,28 @@
+package business
+
+import (
+	"context"
+
+	"mock-payment-provider/primitive"
+)
+
+// Payment interface handles anything to do with the completion of a payment,
+// from the customer's standpoint.
+type Payment interface {
+	// GetDetails will acquire a payment detail from an ID coming from e-money or
+	// virtual account payment.
+	GetDetails(ctx context.Context, id string) (PaymentDetailsResponse, error)
```

Comment:
```suggestion
	GetDetail(ctx context.Context, id string) (PaymentDetailsResponse, error)
```
singular kan? iya. ok